### PR TITLE
Create KeyVault.json

### PR DESCRIPTION
**Pull request by jlichwa**
Event definitions for Key Vault Certificates, Keys and Secrets - NewVersion, Near Expiry and Expired

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@jlichwa Hello. Why do we have this PR? I don't see any path referring these definitions. We cannot pass the Avocado's check.

File: specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+    "title": "Schema of Azure Vault events published to Azure Event Grid",
+
+    "description": "Describes the schema of the Azure Key Vault events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+
+  },
+
+  "paths": {},
+
+  "definitions": {
+
```
Comment (MyronFanQiu) [inline specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json]: Could you please format this file? We don't want too much empty line.

File: specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+    "version": "2018-01-01",
+
+    "title": "Schema of Azure Vault events published to Azure Event Grid",
+
+    "description": "Describes the schema of the Azure Key Vault events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+
+  },
+
+  "paths": {},
```
Comment (MyronFanQiu) [inline specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json]: I cannot see any `paths` defined under this API version. It's quite weird. Is this PR in progress or ready_for_review?

**Review by mmyyrroonn — APPROVED** [review]

File: specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json (hunk 2 above)
Comment (jlichwa) [reply specification/eventgrid/data-plane/Microsoft.KeyVault/stable/2018-01-01/KeyVault.json]: I've been guided by EventGrid team (Justin Conway) to add schema for our new events. I believe after that their team is generating API based on that schema.

**Review by Kishp01 — APPROVED** [review]

**Review by kalyanaj — COMMENTED** [review]
For consistency with the rest of the files under EventGrid dataplane, please consider removing the additional line spacing used in this file.

**Review by kalyanaj — CHANGES_REQUESTED** [review]
For each definition name such as CertificateNewVersionCreated, KeyNearExpiry etc. since all the definitions share the same namespace Microsoft.EventGrid.Models, please add the KeyVault prefix, as an example, please rename CertificateNewVersionCreated to KeyVaultCertificateNewVersionCreated. This will help with better discoverability of all KeyVault specific event definitions. Also, typically, we use the "EventData" suffix for such event definitions (e.g. StorageBlobCreatedEventData), so please consider adding EventData suffix as well to each definition name. (for example "CertificateNewVersionCreated" would become  KeyVaultCertificateNewVersionCreatedEventData.).

**Comment by jlichwa** [comment]
Updated names KeyVault prefix and Event Data Suffix

**Comment by MyronFanQiu** [comment]
@jlichwa Can I assume that this PR is in progress and there will be another team working on the API paths later?

**Comment by jlichwa** [comment]
Yes my team adding schema definition only.
Eventgrid team - @J. Kalyana Sundaram<mailto:kalyanaj@microsoft.com>, would work on API after it is merged.

Jack Lichwa
Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jlichwa** [comment]
@MyronFanQiu Can we Merge based on comments?  @kalyanaj

**Comment by MyronFanQiu** [comment]
@jlichwa I still don't understand the purpose of this PR. Who will use these definitions?

**Comment by jlichwa** [comment]
> @jlichwa I still don't understand the purpose of this PR. Who will use these definitions?

It is for EventGridTeam( @kalyanaj ) as part of their OnBoarding process for new events schema. Based on that schema they will build API (they cannot do it without it) and once it is ready for Public Preview, it will be added to Readme.md as a part of API to customers. 
My understanding is just more about Teams separation, EventGrid customers, provide the schema to EventGrid dev team (which is that json file I added) and based on that EventGrid team build API and enable that events schema definitions to public - so 2 teams contribute to the entire feature.

**Comment by kalyanaj** [comment]
@MyronFanQiu, please see my comment earlier in this PR (https://github.com/Azure/azure-rest-api-specs/pull/6975#discussion_r317749960) for more context on how these definitions will be used. (FYI: @jlichwa).

**Review by kalyanaj — APPROVED** [review]
LGTM.

**Comment by sarangan12** [comment]
@jilchwa One comment. You have created a new JSON file in this PR. But this file is never included in the readme.md file. So, no tests will be run on this file unless you actually include it. I believe the failed tests are really unrelated to your file. (which is a seperate discussion). But, first please include the file reference in the readme file.

**Comment by MyronFanQiu** [comment]
Remove the label `WatiforArmFeedback` since it's related with data-plane

**Comment by MyronFanQiu** [comment]
@jlichwa Hello. Resolve the CI and I can merge this PR.

**Review by mmyyrroonn — APPROVED** [review]

**Pull request merged**